SPIFlashLogger.class.nut

```diff
@@ -224,6 +224,68 @@ class SPIFlashLogger {
         });
     }
 
+    function readSync(index = 1) {
```

Comment:
I see. I was just thinking of >= 0 - forward scanning (0 as for the first element) and -1 for reverse scanning. It may be just me who get used to it, but indexing from 0 seems more natural in most cases.

However I'm not insisting, especially if it makes the implementation less consistent.